set-dotnet-versions.bat

```diff
@@ -1,46 +1,58 @@
-
 REM Set these version variables and run the script to change all the files which need version updates
 
-REM The site extension version
+REM The site extension release version
 set major=0
-set minor=3
-set patch=15
+set minor=4
+set patch=0
 set version_postfix=-prerelease
 
-REM The agent version
+REM Specialized version for development package, increment as necessary for testing
+set development_major=100
+
+REM The agent version to deploy
 set agent_version=7.25.0
 
-REM The dotnet tracer version
-set tracer_version=1.21.1
+REM The dotnet tracer version to deploy
+set tracer_version=1.22.0
 
-REM *************************************************************
-REM All of the below code updates versions in files, do not touch
-REM *************************************************************
+REM **************************************************************************************************************************
+REM All of the below code updates versions in files, do not touch unless you wish to modify the structure of those files
+REM **************************************************************************************************************************
 
 set path_regex=[0-9]+_[0-9]+_[0-9]+
 set path_replacement=%major%_%minor%_%patch%
 
 set gitlab_yml=.gitlab-ci.yml
 powershell -Command "(gc .\%gitlab_yml%) -replace '%path_regex%', '%path_replacement%' | Out-File -encoding ASCII .\%gitlab_yml%"
+echo/
 
 set version_regex=[0-9]+\.[0-9]+\.[0-9]+[\-a-zA-Z]*
 powershell -Command "(gc .\%gitlab_yml%) -replace '%version_regex%.+windows-tracer-home.zip', '%tracer_version%/windows-tracer-home.zip' | Out-File -encoding ASCII .\%gitlab_yml%"
+echo/
+
 set version_regex=[0-9]+\.[0-9]+\.[0-9]+[\-\.a-zA-Z0-9]*
 powershell -Command "(gc .\%gitlab_yml%) -replace 'agent-binaries-%version_regex%-1-x86_64.zip', 'agent-binaries-%agent_version%-1-x86_64.zip' | Out-File -encoding ASCII .\%gitlab_yml%"
+echo/
 
-set nuget_replacement=%major%.%minor%.%patch%%version_postfix%
+set release_nuget=dotnet\Datadog.AzureAppServices.nuspec
+set release_version=%major%.%minor%.%patch%%version_postfix%
+powershell -Command "(gc .\%release_nuget%) -replace '%version_regex%', '%release_version%' | Out-File -encoding ASCII .\%release_nuget%"
 
-set nuget_files=dotnet\Datadog.Development.AzureAppServices.nuspec dotnet\Datadog.AzureAppServices.nuspec
+set dev_nuget=dotnet\Datadog.Development.AzureAppServices.nuspec
+set development_package_version=%development_major%.%minor%.%patch%%version_postfix%
+powershell -Command "(gc .\%dev_nuget%) -replace '%version_regex%', '%development_package_version%' | Out-File -encoding ASCII .\%dev_nuget%"
 
-(for %%f in (%nuget_files%) do (
-	powershell -Command "(gc .\%%f) -replace '%version_regex%', '%nuget_replacement%' | Out-File -encoding ASCII .\%%f"
-   echo/
-)) > set-versions-log.txt
+set application_host_transform=dotnet\content\applicationHost.xdt
```

Comment:
Just to make sure my understanding is correct, will the directory structure between the two packages be the same or does the `applicationHost.xdt` have to be modified for each package?

Reply:
The directory structure will be exactly the same, as well as all other files in the package.
The only difference at this point is the actual nuget package metadata.

Reply:
I have wondered if it's worth running a transform on the xdt for the development package as well, because part of the install process is about unobtrusively dropping in the new binaries without interruption to the web app, so perhaps I should revisit this.
The development version needs to be considered for it to be a true one to one test.

Reply:
If feasible, I'd recommend changing the dir name and the applicationHost for the development package as well, but I don't want to block the current AAS release on it. Let's leave that as a follow-up item so we can get the current beta out to new users

Reply:
The extensions are installed in different directories, it's just what is within those directories is currently the same.

Reply:
Right, so there's nothing blocking them from being installed by side-by-side, but just upgrading the Development package (even with new Development package versions) is not optimal right? Since those upgrades will be in the same directory (from the beta package version)?

Reply:
It would likely be a condition where whichever extension is the last to set the variables would win.